### DISCUSSION TITLE
How to control DAG workflow using REST API

### DISCUSSION THREAD

**Discussion by prasadbasutkar** [apache/airflow#24707] [General]
Hi,

We already have workflow task or milestones scattered across different processes/application some of then from JAVA application and some are from bash script.So our requirement is to have workflow which can run as decoupled independent system with DAG configured with REST api support.DAG run , start of task and end of task should be controlled using REST API from outside airflow application.
I have gone through REST API documentation but didn't find it use case which should only start the DAG in queued state and application will control the flow of task like start of rask and end of task based on milesone events from independent process/script.
Currently when i launch http://wp-p3s-2a:8080/api/v1/dags/ebisearch/dagRuns it start and finishes the workflow, but my requirement is flow of airflow workflow task should be control by REST endpoint calls from another process/application.
Please let me know is it possible to acheive this using apache airflow REST API?

Regards,
Prasad

**Comment by potiuk** [comment]
You need to reverse your thinking. Airlfow DAG has to "poll" for milestone using sensors.

The only way to implement it is to add sensors in your workflow that will poll for status in some external source (database, S3 etc.) you can write your own "controller" to update those "things" that your DAG will be waiting for (for example update a Dtabase entry, create an S3 object etc.). 

This will allow you to implement what you want.

**Reply by prasadbasutkar** [reply]
Many thanks for your valuable feedback , i will read about sensors and how to use them

**Reply by prasadbasutkar** [reply]
@potiuk potiukDo you aware of any REST dag endpoint which will allow us update runtime dag instance by passing conf {"task_complete_name_key","task_name_1"} and in python_callable method we can check that value to complete the task

**Reply by potiuk** [reply]
It's  the same question and same answer. You need to  reverse thinking and call from Airflow workflow to decide when to continue. There is no REST API for that and no other way to do it in Airflow.
